src/mscorlib/src/System/String.cs

```diff
@@ -2128,11 +2120,12 @@ public static int Compare(String strA, String strB, StringComparison comparisonT
                 return 1;
             }
 
-            if (!(value is String)) {
+            var asString = value as string;
+            if (asString == null) {
```

Comment:
This is also not helping anything - if the optimizations work as expected.

The assertion propagation optimization in the JIT will know that the value is a string, and it will eliminate the string cast a few lines later.